src/main/java/it/pagopa/pm/gateway/service/VposDeleteService.java

```diff
@@ -0,0 +1,159 @@
+            refundOutcome = executeOrderStatus(entity, stepZeroRequest);
+            if (refundOutcome.equals(OK)) {
+                refundOutcome = executeRevert(entity, stepZeroRequest);
+                if(!refundOutcome.equals(OK)) {
+                    return createDeleteResponse(requestId, entity.getErrorMessage(), refundOutcome, entity);
+                }
+            } else {
+                return createDeleteResponse(requestId, entity.getErrorMessage(), refundOutcome, entity);
+            }
+        } catch (Exception e) {
+            log.error(GENERIC_ERROR_MSG + entity.getIdTransaction() + e);
+            return createDeleteResponse(requestId, GENERIC_REFUND_ERROR_MSG + requestId, KO, entity);
+        }
+        return createDeleteResponse(requestId, null, refundOutcome, entity);
+    }
+
+    private StepZeroRequest getStepZeroRequest(PaymentRequestEntity entity) throws JsonProcessingException {
+        StepZeroRequest stepZeroRequest = objectMapper.readValue(entity.getJsonRequest(), StepZeroRequest.class);
+        stepZeroRequest.setIsFirstPayment(false);
+        return stepZeroRequest;
+    }
+
+    private RefundOutcome executeOrderStatus(PaymentRequestEntity entity, StepZeroRequest stepZeroRequest) throws IOException {
+        log.info("Calling VPOS - OrderStatus - for requestId: " + entity.getGuid());
+        Map<String, String> params = vPosRequestUtils.buildOrderStatusParams(stepZeroRequest, entity.getCorrelationId());
+        HttpClientResponse clientResponse = callVPos(params);
+        VposOrderStatusResponse response = vPosResponseUtils.buildOrderStatusResponse(clientResponse.getEntity());
+        return checkOrderStatusResultCode(response, entity);
+    }
+
+    private RefundOutcome executeRevert(PaymentRequestEntity entity, StepZeroRequest stepZeroRequest) throws IOException {
+        log.info("Calling VPOS - Revert - for requestId: " + entity.getGuid());
+        Map<String, String> params = vPosRequestUtils.buildRevertRequestParams(stepZeroRequest, entity.getCorrelationId());
+        HttpClientResponse clientResponse = callVPos(params);
+        AuthResponse response = vPosResponseUtils.buildAuthResponse(clientResponse.getEntity());
+        return checkRevertResultCode(response, entity);
+    }
+
+    private VposDeleteResponse createDeleteResponse(String requestId, String errorMessage, RefundOutcome refundOutcome, PaymentRequestEntity entity) {
+        VposDeleteResponse response = new VposDeleteResponse();
+        response.setRequestId(requestId);
+        response.setError(errorMessage);
+        response.setRefundOutcome(refundOutcome.name());
+        if(Objects.nonNull(entity)) {
+            response.setStatus(entity.getStatus());
+        }
+
+        log.info("END - Vpos refund for requestId: " + requestId);
+        return response;
+    }
+
+    private HttpClientResponse callVPos(Map<String, String> params) throws IOException {
```

Comment:
There isn't any check on valid params and their format. Is this wanted?

Reply:
It is imported from PM.
The check is done during the params construction